src/libYARP_OS/src/Thread.cpp

```diff
@@ -59,8 +59,8 @@ Thread::Thread() :
 
 
 Thread::~Thread() {
-    ((ThreadImpl*)implementation)->close();
-    if (implementation!=YARP_NULLPTR) {
+    if (implementation != YARP_NULLPTR) {
```

Comment:
@damn1 I had some strange behaviours while testing pointers agains `0`/`NULL`/`nullptr` (see https://stackoverflow.com/questions/11279715/nullptr-and-checking-if-a-pointer-points-to-a-valid-object). Perhaps it is better to just have `if (implementation)`.

Reply:
That I like more anyway 🤣

Reply:
I think that the strange behaviour might happen when the pointer is not properly initialized, I don't think this is the case... Also we have it everywhere in YARP.
I don't have a preference on the matter, so for me it is ok in both ways.